lib/generators/administrate/dashboard/dashboard_generator.rb

```diff
@@ -53,9 +53,16 @@ def namespace
       end
 
       def attributes
-        klass.reflections.keys +
-          klass.columns.map(&:name) -
-          redundant_attributes
+        columns = klass.columns.map(&:name)
+
+        (klass.reflections.keys +
+          columns -
+          redundant_attributes -
+          %w[id created_at updated_at]).sort.tap do |result|
+            result.unshift(klass.primary_key) if klass.primary_key.present?
+            result.push("created_at") if columns.include?("created_at")
+            result.push("updated_at") if columns.include?("updated_at")
+          end
```

Comment:
Responding to @nickcharlton's comment, yeah, this sort of stuff is always tricky and open to personal preference. This code doesn't offend me, but I do wonder if it's possible to make it more readable... How about the following?

```suggestion
        unsorted_attrs = (klass.reflections.keys +
          klass.columns.map(&:name) -
          redundant_attributes)

        primary_key = unsorted_attrs.delete(klass.primary_key)
        updated_at_key = unsorted_attrs.delete("updated_at")
        created_at_key = unsorted_attrs.delete("created_at")

        (
          [primary_key] +
          unsorted_attrs.sort +
          [created_at_key, updated_at_key]
        ).compact
```

I'm not married to my proposed solution either, just checking what others think :slightly_smiling_face:

Reply:
@pablobm thanks for the suggestion. I've taken what you proposed and modified it slightly. Let me know what you think .